src/irc/ClientPool.ts

```diff
@@ -39,30 +39,23 @@ interface ReconnectionItem {
  * and may be closed for a variety of reasons.
  */
 export class ClientPool {
-    private botClients: { [serverDomain: string]: BridgedClient|undefined};
+    private botClients: Map<string, BridgedClient>;
     private virtualClients: { [serverDomain: string]: {
-        nicks: { [nickname: string]: BridgedClient|undefined};
-        userIds: { [userId: string]: BridgedClient|undefined};
-        pending: { [nick: string]: BridgedClient};
+        nicks: Map<string, BridgedClient>;
+        userIds: Map<string, BridgedClient>;
+        pending: Map<string, BridgedClient>;
     };};
-    private virtualClientCounts: { [serverDomain: string]: number };
     private reconnectQueues: { [serverDomain: string]: QueuePool<ReconnectionItem> };
     private identGenerator: IdentGenerator;
     private ipv6Generator: Ipv6Generator;
     private ircEventBroker: IrcEventBroker;
     constructor(private ircBridge: IrcBridge, private store: DataStore) {
         // The list of bot clients on servers (not specific users)
-        this.botClients = { };
+        this.botClients = new Map();
 
         // list of virtual users on servers
         this.virtualClients = { };
 
-        // map of numbers of connected clients on each server
-        // Counting these is quite expensive because we have to
-        // ignore entries where the value is undefined. Instead,
-        // just keep track of how many we have.
-        this.virtualClientCounts = { };
```

Comment:
Does this no longer need to be initialised?

Reply:
Solved by https://github.com/matrix-org/matrix-appservice-irc/pull/1034/files#r409475606